src/FSharp.Literate/Transformations.fs

```diff
@@ -324,30 +324,37 @@ module Transformations =
-                    let lines = code.Trim('\r', '\n').Replace("\r\n", "\n").Replace("\n\r", "\n").Replace("\r", "\n").Split('\n')
-                    let numberLength = lines.Length.ToString().Length
-                    let linesLength = lines.Length
-                    writer.Write("<td class=\"lines\"><pre class=\"fssnip\">")
-                    for index in 0..linesLength-1 do
-                      let lineStr = (index + 1).ToString().PadLeft(numberLength)
-                      writer.WriteLine("<span class=\"l\">{0}: </span>", lineStr)
-                    writer.Write("</pre>")
-                    writer.WriteLine("</td>")
-
-                  writer.Write("<td class=\"snippet\">")
-                  
-                  match SyntaxHighlighter.FormatCode(lang, code) with
-                  | true, code -> Printf.fprintf writer "<pre class=\"fssnip highlighted\"><code lang=\"%s\">%s</code></pre>" lang code
-                  | false, code -> Printf.fprintf writer "<pre class=\"fssnip\"><code lang=\"%s\">%s</code></pre>" lang code
-
-                  writer.Write("</td></tr></table>")
+                  if String.IsNullOrEmpty prismClasses then
+                      writer.Write("<table class=\"pre\">")
+                      writer.Write("<tr>")
+                      if ctx.GenerateLineNumbers then 
+                        // Split the formatted code into lines & emit line numbers in <td>
+                        // (Similar to formatSnippets in FSharp.CodeFormat\HtmlFormatting.fs)
+                        let lines = code.Trim('\r', '\n').Replace("\r\n", "\n").Replace("\n\r", "\n").Replace("\r", "\n").Split('\n')
+                        let numberLength = lines.Length.ToString().Length
+                        let linesLength = lines.Length
+                        writer.Write("<td class=\"lines\"><pre class=\"fssnip\">")
+                        for index in 0..linesLength-1 do
+                          let lineStr = (index + 1).ToString().PadLeft(numberLength)
+                          writer.WriteLine("<span class=\"l\">{0}: </span>", lineStr)
+                        writer.Write("</pre>")
+                        writer.WriteLine("</td>")
+
+                      writer.Write("<td class=\"snippet\">")
+                      
+                      match SyntaxHighlighter.FormatCode(lang, code) with
+                      | true, code -> Printf.fprintf writer "<pre class=\"fssnip highlighted\"><code lang=\"%s\">%s</code></pre>" lang code
+                      | false, code -> Printf.fprintf writer "<pre class=\"fssnip\"><code lang=\"%s\">%s</code></pre>" lang code
+
+                      writer.Write("</td></tr></table>")
+                  else
+                      // prism support
+                      let code = Manoli.Utils.CSharpFormat.SourceFormat.EscapeHtml(code, 4)
+                      Printf.fprintf writer "<pre class=\"%s\"><code lang=\"%s\">%s</code></pre>" prismClasses lang code
```

Comment:
Yes indeed that was an oversight, but there is still another issue: This PR disables FCS-snippet evaluation, but basically that was used to "detect" if the snippet is valid F# code, after this PR (when using the environment variables) lots of boxes have no language at all which is very very unfortunate (especially since this is all F# code and this is FSharp.Formatting)

So I'm trying to refactor the code in a way to only detect if code is F# but holy crap that's not easy. Fallback is to have an environment variable define a "fallback" language to be used when no other language is specified. But this is all a big hack and I don't like any of this.

The proper fix would be to embrace FCS in addition to prism (no idea how), this way we could have very nice code in the web

Reply:
Prism works fine with just `<pre><code class="lang-****">...</code></pre>` so you can skip the prism class in `<pre>`. That way it should also work with highlight js or similair tools.

Have a look at https://github.com/fsharp/FAKE/blob/release/next/help/markdown/fake-debugging.md for example.
This is pretty much the reason why we have to use html there instead of just the default markdown (e.g ```json ... ```).